pkg/platform/platform.go

```diff
@@ -6,28 +6,38 @@ import (
 
 	configv1 "github.com/openshift/api/config/v1"
 	"github.com/openshift/cluster-network-operator/pkg/bootstrap"
+	cnoclient "github.com/openshift/cluster-network-operator/pkg/client"
 
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	types "k8s.io/apimachinery/pkg/types"
-	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var cloudProviderConfig = types.NamespacedName{
 	Namespace: "openshift-config-managed",
 	Name:      "kube-cloud-config",
 }
 
-func BootstrapInfra(kubeClient crclient.Client) (*bootstrap.InfraBootstrapResult, error) {
+func InfraStatus(client cnoclient.Client) (*bootstrap.InfraStatus, error) {
 	infraConfig := &configv1.Infrastructure{}
-	if err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
+	if err := client.Default().CRClient().Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
 		return nil, fmt.Errorf("failed to get infrastructure 'cluster': %v", err)
 	}
 
-	res := &bootstrap.InfraBootstrapResult{
+	res := &bootstrap.InfraStatus{
 		PlatformType:         infraConfig.Status.PlatformStatus.Type,
 		PlatformStatus:       infraConfig.Status.PlatformStatus,
 		ExternalControlPlane: infraConfig.Status.ControlPlaneTopology == configv1.ExternalTopologyMode,
+		APIServers:           map[string]bootstrap.APIServer{},
+	}
+
+	// Extract apiserver URLs from the kubeconfig(s) passed to the CNO
```

Comment:
note: allow overriding the "default" cluster here. Probably via an environment variable. This is needed for Hypershift.

Reply:
This is now done.